addon/mixins/run.js

```diff
@@ -88,18 +98,27 @@ export default Mixin.create({
       !this.isDestroyed
     );
 
-    let pendingTimers = getOrAllocate(this, '_pendingTimers', Array);
+    let registeredTimers = getPendingTimers(this);
+
+    if (!this._isRunTaskDisposableRegistered) {
+      registerDisposable(this, () => {
+        for (let i = 0; i < registeredTimers.length; i++) {
```

Comment:
So I think this will close over `registeredTimers`, which may cause a leak. Thoughts @stefanpenner @rwjblue?